helper/schema/schema.go

```diff
@@ -2353,6 +2384,36 @@ func (m schemaMap) validateType(
 	return diags
 }
 
+// hasWriteOnly returns true if the schemaMap contains any WriteOnly attributes.
```

Comment:
One additional usage of this `hasWriteOnly` I think we missed is applying this to managed resource blocks that are not write-only themselves. For example, ATM, this schema is considered valid for a managed resource:

```go
map[string]*schema.Schema{
	"nested_list_block": {
		Type:     schema.TypeList,
		Optional: true,
		Computed: true,
		Elem: &schema.Resource{
			Schema: map[string]*schema.Schema{
				"writeonly_string": {
					Type:      schema.TypeString,
					Required:  true,
					WriteOnly: true,
				},
			},
		},
	},
},
```

Reply:
In the RFC, this specific rule was: `Block types with Computed set to true cannot contain WriteOnly attributes`